Automattic/portkey-android
# [Build] Update Kotlin to 1.6.10

This PR is a prerequisite for the `Gradle 8.1.1 & AGP 8.0.2 Upgrade for WPAndroid, WCAndroid & Related Libs` project.

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

-----

This PR upgrades Kotlin to [1.6.10](https://github.com/JetBrains/kotlin/releases/tag/v1.6.10).

FYI: This is primarily done to unblock development with JDK17.

PS: It also includes the following: [Remove unnecessary kotlin stdlib dependency from all modules](https://github.com/Automattic/stories-android/commit/eaa9d3adc2bb7caef14056d37d1d8fb518598d8c)

-----

Lint Fix List:

1. [Fix lint build warning on file not existing](https://github.com/Automattic/stories-android/commit/b5346d9d8582080aea68df75b1f5897f82f3ee52)
2. [Fix lint build failure on could not find org wordpress lint](https://github.com/Automattic/stories-android/commit/c658346458f92d098e803d2fe15ccd8fb0e0bea7)
3. [Fix lint build failure on androidx test espresso manifest merger](https://github.com/Automattic/stories-android/commit/be74838e861642b8836066f95d614d353a367ca1)
4. [Add lint check to buildkite linters group](https://github.com/Automattic/stories-android/commit/b7d5b4624e2597426a8708055acfface7523a14a)

Test Restructure List:

1. [Remove all ui test related configuration from project](https://github.com/Automattic/stories-android/commit/8e9b6826742de102f92e1c2548c17d207606edc0)
2. [Remove unnecessary unit test related configuration from project](https://github.com/Automattic/stories-android/commit/d7ecea842528f1eed467d1eac0fc0e5af422dd5a)
3. [Update junit to 4.13.2](https://github.com/Automattic/stories-android/commit/29b653488c9568d0c3d3899ab0f73cf09ea1b4c1)
4. [Remove unnecessary kotlin test junit dependency](https://github.com/Automattic/stories-android/commit/2f8e732e3b2c156dbbdcf58aa9d2e2c5c9013762)

Dependency Versions Refactor List:

1. [Extract all sdk version to a separate ext block](https://github.com/Automattic/stories-android/commit/d25ecd52421da1492ce281ab3e97ca881c89f6c6)
2. [Rename lifecycle to androidx lifecycle version](https://github.com/Automattic/stories-android/commit/cfabe344bc455f3b0c09e6a3497bcae6d9fd3cf8)
3. [Rename coroutines to kotlinx coroutines version](https://github.com/Automattic/stories-android/commit/592e4c569e16b89953353cd7bd2c315a4618ef39)
4. [Rename constraint layout to androidx constraint layout version](https://github.com/Automattic/stories-android/commit/f61e0b72923a058a1a00fcb34fbc7485738f7a37)
5. [Rename appcompat to androidx appcompat version](https://github.com/Automattic/stories-android/commit/956847d7bbe2d87640f6a4a7ec2dde7d2e1c2a43)
6. [Rename core to androidx core version](https://github.com/Automattic/stories-android/commit/1b8044bcfbf3b8ab9e8e583ab10b8a7f992ca5a4)
7. [Rename navigation to androidx navigation version](https://github.com/Automattic/stories-android/commit/d03c432c6a280215a92a5fe5c27fadc223e337f3)
8. [Rename serialization to kotlinx serialization version](https://github.com/Automattic/stories-android/commit/33475fe69ed15f1c7062d3e41451c7745656f14b)
9. [Extract glide version to root build gradle](https://github.com/Automattic/stories-android/commit/a2e1efd26b94136ca4945a8e5640456576aa6e8b)
10. [Extract glide transformations version to root build gradle](https://github.com/Automattic/stories-android/commit/552c2cd01d463fe2c6500024d333cf34fa6b7022)
11. [Extract wordpress lint version to root build gradle](https://github.com/Automattic/stories-android/commit/16954b115c872a94654646b058488796538de364)
12. [Extract androidx camera version to root build gradle](https://github.com/Automattic/stories-android/commit/df8b2f433c207782557cf435973d9c558fa621ec)
13. [Extract google material version to root build gradle](https://github.com/Automattic/stories-android/commit/1ba4fbf95f8f7621856ade5ad70a70903b0a72f6)
14. [Extract chrisbanes photoview version to root build gradle](https://github.com/Automattic/stories-android/commit/7a93d4b2d98f6989985659ce1b1d492bdadfeab8)
15. [Extract androidx emoji version to root build gradle](https://github.com/Automattic/stories-android/commit/ee3640571118a85723535d14e051b7a0f1f092ac)
16. [Extract androidx recyclerview version to root build gradle](https://github.com/Automattic/stories-android/commit/946116cb6d482c57a41ae298a4a4af782cabe5c3)
17. [Extract event bus version to root build gradle](https://github.com/Automattic/stories-android/commit/8e1352a6d6091bdc93ec155d7ac0f64bd4a2ae99)
18. [Extract junit version to root build gradle](https://github.com/Automattic/stories-android/commit/7973312eede351b0e52cac89461526d1aa389fba)
19. [Extract androidx preference version to root build gradle](https://github.com/Automattic/stories-android/commit/92e5ac4af6b61e98ecb3a011b86b12d09d9beeb1)
20. [Remove duplicate androidx preference ktx dep from app module](https://github.com/Automattic/stories-android/commit/5d34148b192879c7bb0414a30fe545c92bf3ed2e)
21. [Extract sentry android version to root build gradle](https://github.com/Automattic/stories-android/commit/3bdb297349785b01ab024e1e2138ebfaf3db1cbc)
22. [Extract sentry version to settings build gradle](https://github.com/Automattic/stories-android/commit/883ef6156f8740eb8ac86edfc8426e25fe7b397c)
23. [Extract automattic publish to s3 version to settings build gradle](https://github.com/Automattic/stories-android/commit/c2da7adf7e25c0a131f12c5d6a73362e0e970315)
24. [Extract slf4j version to root build gradle](https://github.com/Automattic/stories-android/commit/675dd6b2abb5485e16df8260e2f7148999ba51af)
25. [Extract debug versions to root build gradle](https://github.com/Automattic/stories-android/commit/1c0c815a4f0750931bf08e90cbc6f6ff71809c19)
26. [Add versions groups to root build gradle](https://github.com/Automattic/stories-android/commit/6525a1eca5f79783a8d4ec5595155e211c620f44)

-----

## To test

1. Verify that all the CI checks are successful (especially the newly added `Lint` check).
2. Smoke test the `app`.
3. Test for build time increases (`Local` & `CI`).
